### DISCUSSION TITLE
Is there a way to change PositionZ?

### DISCUSSION THREAD

**Discussion by msspad** [sn4k3/UVtools#295] [Q&A] [answered]
By definition, bottom layers have a much higher ExposureTime. That leads not only to an elephant foot in xy direction, but bottom layers also grow higher than LayerHeight. This higher height results in ugly compressed layers at the bottom. Is there a way to change PositionZ to get rid of these compressed layers? or any similar method?

For example, LayerHeight=0.05 mm (originating from (Prusa)Slicer), NumBottomLayers=2:
- layer 0 bottom: LayerHeight=0.05, PositionZ=0.05
- layer 1 bottom: LayerHeight=0.05, manually change to PositionZ=0.15mm
- layer 2: LayerHeight=0.05, manually change to PositionZ=0.25
- layer 3: LayerHeight=0.05, recalculate PositionZ=0.3
- layer 3: LayerHeight=0.05, recalculate PositionZ=0.35
Nothing else changes in those layers. The total height will theoretically grow by 0.1mm, but the model should show a much nicer elephant foot.

**Comment by sn4k3** [comment, marked as answer]
ATM only via Scripting.
Some people have requested this. I will see how i can integrate this. But keep in mind that change layer position is not very safe and invalidate model height. (If affect zone is raft then is not a problem)

**Reply by msspad** [reply]
ok, thanks for hinting at scripting, will check that out.
yeah, the model grows in height but I have so many cases where nicer looking is much more important than dimension accuracy. It could be properly solved in slicers but we should not wait for those... In any case, most features in UVTools are for advanced users and you should anyway know what you are doing (or at least be prepared to learn).
